src/stedi/cdk/lambda.clj

```diff
@@ -1,37 +1,28 @@
 (ns stedi.cdk.lambda
-  (:require [clojure.string :as string]
+  (:require [clojure.java.io :as io]
+            [clojure.string :as string]
             [stedi.cdk :as cdk]
-            [stedi.cdk.lambda.build :as build]))
+            [stedi.lambda.build :as lambda-build]))
 
 (cdk/import ["@aws-cdk/core" Duration]
-            ["@aws-cdk/aws-lambda" Function AssetCode Runtime LayerVersion])
+            ["@aws-cdk/aws-lambda" Code Runtime Function])
 
-(defn Clj
-  "Wraps `@aws-cdk/aws-clambda.Function` to be clojure friendly.
-
-    fn          - A var pointing to a handler function.
-    environment - A map of environment variables
-    handler     - Only necessary if fn isn't specified
-    aot         - A list of namespaces to AOT compile
-  "
-  [parent id {:keys [fn environment handler aot] :as props}]
-  (let [{:keys [lib-layer src]} (build/build aot)
-
-        env       (merge environment
-                         (when fn {"STEDI_LAMBDA_HANDLER" (-> fn symbol str)}))
-        function  (Function parent id
-                            (merge {:code        (AssetCode src)
-                                    :handler     (or handler "stedi.lambda.entrypoint::handler")
-                                    :runtime     (:JAVA_8 Runtime)
-                                    :environment env
-                                    :memorySize  2048
-                                    :timeout     (Duration/minutes 1)}
-                                   (dissoc props :environment :aot :fn)))
-        lib-layer (LayerVersion function "lib-layer" {:code (AssetCode lib-layer)})]
-    (Function/addLayers function lib-layer)
-    function))
-
-(defn ^:deprecated clj
-  "Deprecated in favor of `stedi.cdk.lambda/Clj`."
-  [_ parent id {:keys [fn environment handler aot] :as props}]
-  (Clj parent id props))
+(defn fn-from-var
+  "Creates a @aws-cdk/aws-lambda.Function from a Clojure var. Includes
+  sane defaults that can be overridden by passing in optional
+  `function-props`."
+  ([scope id var]
+   (fn-from-var scope id var {}))
+  ([scope id var function-props]
```

Comment:
This is a general comment and need not be addressed. 

I understand Clojure will treat `var` in this context as an argument and not as the core function. Nonetheless, I have started to avoid shadowing core functions when naming arguments for readability and speed of understanding, e.g., a person need not think "Is that the var function? Oh, it's just an argument shadowing the core function." In other words, by doing so, we do away-with yet another small but nonetheless present mental tax.

WDYT?

Reply:
In general I avoid when there exists a better alternative but I will shadow when it clarifies intention and the function being shadowed isn't commonly used.